internal/middleware/middleware.go

```diff
@@ -0,0 +1,148 @@
+package middleware
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/PereRohit/util/log"
+	"github.com/PereRohit/util/response"
+	"github.com/dgrijalva/jwt-go"
+	"github.com/vatsal278/TransactionManagementService/internal/codes"
+	svcCfg "github.com/vatsal278/TransactionManagementService/internal/config"
+	"github.com/vatsal278/TransactionManagementService/internal/model"
+	"github.com/vatsal278/TransactionManagementService/internal/repo/authentication"
+	"github.com/vatsal278/TransactionManagementService/pkg/session"
+	"github.com/vatsal278/go-redis-cache"
+	"net/http"
+	"strings"
+)
+
+type TransactionMgmtMiddleware struct {
+	cfg    *svcCfg.Config
+	jwt    authentication.JWTService
+	cacher redis.Cacher
+}
+
+type respWriterWithStatus struct {
+	status   int
+	response string
+	http.ResponseWriter
+}
+
+func (w *respWriterWithStatus) WriteHeader(code int) {
+	w.status = code
+	w.ResponseWriter.WriteHeader(code)
+}
+
+func (w *respWriterWithStatus) Write(d []byte) (int, error) {
+	w.response = string(d)
+	return w.ResponseWriter.Write(d)
+}
+
+func NewTransactionMgmtMiddleware(cfg *svcCfg.SvcConfig) *TransactionMgmtMiddleware {
+	return &TransactionMgmtMiddleware{
+		cfg:    cfg.Cfg,
+		jwt:    cfg.JwtSvc.JwtSvc,
+		cacher: cfg.Cacher.Cacher,
+	}
+}
+func (u TransactionMgmtMiddleware) ExtractUser(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		cookie, err := r.Cookie("token")
+		if err != nil {
+			log.Error(err)
+			response.ToJson(w, http.StatusUnauthorized, codes.GetErr(codes.ErrUnauthorized), nil)
+			return
+		}
+		if cookie.Value == "" {
+			log.Error(err)
```

Comment:
Remove this log for error as this will be nil.

Reply:
done